### ISSUE TITLE
Allow for variable bindings (let statements) inside a template

### ISSUE THREAD

**Issue by sackery** [lfairy/maud#57]
For example:
`@{
let a = bit.source.unwrap();
}`

**Comment by lfairy** [comment]
Can you give me an example of how such a feature would be used?

I've considered it in the past, but held back because I couldn't find a use case for it.

Note that you can declare variables [outside the `html!` block](https://github.com/lfairy/karkinos/blob/477808ba8363dd8401e9e10a505b62b2d4314491/src/views.rs#L110) as well.

**Comment by sackery** [comment]
because some situation e.g. @for n in 1..2 {}.
we want to insert normal rust expression to cache something.

**Comment by sackery** [comment]
```rust
@for n in somethings {
    div {
        @ {
            let item = n.something.unwrap(); // here item in something is move out, so we cannot unwrap twise.
        }
        h1 { (item.name) }
        a { (item.age) }
    }
}
```

**Comment by lfairy** [comment]
That sounds like a reasonable feature to ask for. I have a feeling we need to go all the way to arbitrary Rust statements to solve your problem though. Maybe a `@let` special form?

I'll need to think about this a bit more.

**Comment by sackery** [comment]
but maybe this is powerful and smart.
```rust
html! {
    @ {
        somefunction(); // I can run any rust code here.
    }
}

**Comment by TheNeikos** [comment]
I have to say that I didn't need to do this so far, anything that should be run once can be run outside the macro, and the `@if let` syntax allows for the rest

**Comment by lfairy** [comment]
Implemented in 39602a075cfc14fe86964fcd2a119ce97f355a3b. The new `@let` form is pretty much like `@if let`, except the pattern cannot fail.

@sackery I don't want to allow arbitrary Rust statements within a template. If your view logic is complicated enough that you need a whole block for it, then that's a sign that it should be in a separate function altogether.
